pkg/autoscaler/autoscaler_test.go

```diff
@@ -113,3 +114,141 @@ func TestAutoscaler_Awake(t *testing.T) {
 		return latestRequest != 1
 	}, downscaleStabilizationPeriod, time.Second)
 }
+
+func TestAutoscaler_MinReplicas(t *testing.T) {
+	t.Parallel()
+	log := newLogger(t)
+
+	mux := sync.RWMutex{}
+	var latestRequest int32
+
+	minReplicas := int32(5)
+	maxReplicas := int32(10)
+
+	scalerMock := &ScalerFunc{
+		ScaleFunc: func(apiName string, request int32) error {
+			mux.Lock()
+			defer mux.Unlock()
+
+			latestRequest = request
+			return nil
+		},
+		GetInFlightRequestsFunc: func(apiName string, window time.Duration) (*float64, error) {
+			return pointer.Float64(0), nil
+		},
+		GetAutoscalingSpecFunc: func(apiName string) (*userconfig.Autoscaling, error) {
+			return &userconfig.Autoscaling{
+				MinReplicas:    minReplicas,
+				MaxReplicas:    maxReplicas,
+				InitReplicas:   minReplicas,
+				TargetInFlight: pointer.Float64(1),
+				Window:         time.Second,
+			}, nil
+		},
+		CurrentReplicasFunc: func(apiName string) (int32, error) {
+			return minReplicas + 1, nil
+		},
+	}
+
+	autoScaler := &Autoscaler{
+		logger:              log,
+		crons:               make(map[string]cron.Cron),
+		scalers:             make(map[userconfig.Kind]Scaler),
+		lastAwakenTimestamp: make(map[string]time.Time),
+	}
+	autoScaler.AddScaler(scalerMock, userconfig.RealtimeAPIKind)
+
+	api := userconfig.Resource{
+		Name: "test",
+		Kind: userconfig.RealtimeAPIKind,
+	}
+
+	autoscaleFn, err := autoScaler.autoscaleFn(api)
+	require.NoError(t, err)
+
+	ticker := time.NewTicker(500 * time.Millisecond)
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				err := autoscaleFn()
+				require.NoError(t, err)
+			}
+		}
+	}()
+
+	require.Never(t, func() bool {
+		mux.RLock()
+		defer mux.RUnlock()
+		return latestRequest < minReplicas
+	}, 3*time.Second, time.Second)
```

Comment:
What do you say about running this for longer?

Reply:
it doesn't really matter, because the stabilization periods are zero. we want to keep unit tests as short as possible

Reply:
reduced the interval each autoscaleFn runs to have it run longer instead